### DISCUSSION TITLE
Is there an example demonstrating how to create a self-contained installer ?

### DISCUSSION THREAD

**Discussion by jcfr** [Kitware/trame#33] [Q&A]
More specifically, I am looking for example of GitHub action and projects allowing to do the `Desktop` part mentioned here:

![image](https://user-images.githubusercontent.com/219043/150418146-ec1c9e91-8a25-4eaa-9065-48047177e7b9.png)

**Comment by jourdain** [comment]
In order to start your app as a desktop application, you just need to call [`layout.start_desktop_window()`](https://github.com/Kitware/cookiecutter-trame/blob/master/%7B%7Bcookiecutter.package_name%7D%7D/%7B%7Bcookiecutter.import_name%7D%7D/app/main.py#L10) rather than `layout.start()`. This also imply that you need to install [`pywebview`](https://pywebview.flowrl.com/).

**Comment by jourdain** [comment]
Then you can use `py2app` or something similar for bundling your app in a standalone "App bundle". 
You can see an experiment of it [here](https://github.com/XAITK/xaitk-saliency-web-demo/blob/master/app-mac-setup.py) for Mac and our XAI base app.
